cmd/algorand-indexer/daemon_test.go

```diff
@@ -102,3 +107,85 @@ func TestReadGenesis(t *testing.T) {
 	assert.Contains(t, err.Error(), "readGenesis() err: EOF")
 
 }
+
+func createTempDir(t *testing.T) string {
+	dir, err := os.MkdirTemp("", "indexer")
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	return dir
+}
+
+// Make sure we output and return an error when both an API Config and
+// enable all parameters are provided together.
+func TestConfigWithEnableAllParamsExpectError(t *testing.T) {
+	indexerDataDir := createTempDir(t)
+	defer os.RemoveAll(indexerDataDir)
+	daemonConfig := newDaemonConfig()
+	cmd := newDaemonCmd(daemonConfig)
+	daemonConfig.flags = cmd.Flags()
```

Comment:
This seems a little weird, I guess it's simulating some implementation detail in the Execute function? I'm not sure we see a lot of benefit from testing at the Cmd level vs calling runDaemon directly.

Reply:
Yeah it just needs to be there because `runDaemon` tries to access the flags so we need a non null for `daemonCmd.flags`. I've removed it by setting the variable directly.